src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityAnalyzer.cs

```diff
@@ -173,6 +183,54 @@ SimpleNameSyntax syntax
 			}
 		}
 
+		private static void AnalyzeConditionalImmutability(
+			SyntaxNodeAnalysisContext ctx,
+			SyntaxNode syntax
+		) {
+			// Ignore references in DocComments such as crefs
+			if( syntax.IsFromDocComment() ) {
+				return;
+			}
+
+			// Methods can be either of the following depending on where they are declared
+			MethodDeclarationSyntax methodSyntax = null;
+			LocalFunctionStatementSyntax functionSyntax = null;

```

Comment:
Since the only usage of these is in the expression ` methodSyntax?.TypeParameterList ?? functionSyntax?.TypeParameterList` it would be simpler to instead have a `TypeParameterListSyntax` and use the switch to pull it out.

Basically there's some duplication here: the `switch` is handling the difference between `MethodDeclarationSyntax` and `LocalFunctionStatementSyntax`, but then it has to get handled a second time in the expression (which is harder to read IMO because of multiple `?.` and a `??`).